WcaOnRails/app/models/competition.rb

```diff
@@ -107,7 +108,7 @@ def delegates_must_be_delegates
     end
   end
 
-  def warnings
+  def warnings_for(user)
```

Comment:
On the Rails teams I've worked on, generating HTML in a model is deeply frowned upon, since that belongs in the View part of the app.

Under that philosophy, this code belongs in CompetitionsHelper. I think you can also more easily write it without dealing with `html_safe` over there.